app/assets/javascripts/resources.js

```diff
@@ -29,6 +29,55 @@ function scrollToTop(event) {
   $('html, body').animate({scrollTop: 0}, 'slow');
 }
 
+function getCookie(cname) {
+    var name = cname + "=";
+    var ca = document.cookie.split(';');
+    for(var i = 0; i <ca.length; i++) {
+        var c = ca[i];
+        while (c.charAt(0)==' ') {
+            c = c.substring(1);
+        }
+        if (c.indexOf(name) == 0) {
+            return c.substring(name.length,c.length);
+        }
+    }
+    return "";
+}
+
+function setCookie(cname, cvalue, exdays) {
+    var d = new Date();
+    d.setTime(d.getTime() + (exdays*24*60*60*1000));
+    var expires = "expires="+d.toUTCString();
+    document.cookie = cname + "=" + cvalue + "; " + expires;
+}
+
+function processLike(event, target){
+  event.preventDefault();
+  resourceID = target.parentNode.parentNode.action.split('/')[4]
+ 
+  //create shescoding_likes cookie if it does not exist
+  if (getCookie("_shescoding_likes") === ""){
+    document.cookie = "_shescoding_likes  = {}";
+    newValue = JSON.stringify({[resourceID]: true});
+    setCookie("_shescoding_likes", newValue, 365);
+    incrementLike(event, target); 
+  } 
+
+  //if shescoding_likes cookie does exist
+  else {
+  //check if resource id exists
```

Comment:
Make sure indentations in this function are right - e.g., everything in this else should be indented further.